From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Evan Stade <evanstade@microsoft.com>
Date: Thu, 17 Sep 2026 16:06:56 -0700
Subject: [M152] Convert DCHECK before downcasting DOMArrayBufferBase to CHECK

Original change's description:
> Convert DCHECK before downcasting DOMArrayBufferBase to CHECK
>
> There are certainly other DCHECKs in this general area which may also be
> better as CHECKs, but they're less obviously problematic, so they are
> likely better off going through the NotFatalUntil pipeline.
>
> Bug: 558764482
> Change-Id: Ia1e9e893fb419c5e346276933dc5ea4e132a1122
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8395471
> Commit-Queue: Evan Stade <evanstade@microsoft.com>
> Reviewed-by: Daniel Cheng <dcheng@chromium.org>
> Cr-Commit-Position: refs/heads/main@{#1696699}

(cherry picked from commit b0365badbc09c4809cfc270bdf392d07dace3a9e)

Bug: 562783629,558764482
Change-Id: Ia1e9e893fb419c5e346276933dc5ea4e132a1122
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8419954
Auto-Submit: chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Commit-Queue: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/branch-heads/7977@{#2687}
Cr-Branched-From: b7fe14017379ddffae396d944fb8b59a5896c261-refs/heads/main@{#1669021}

diff --git a/third_party/blink/renderer/core/typed_arrays/dom_array_buffer_view.h b/third_party/blink/renderer/core/typed_arrays/dom_array_buffer_view.h
index d12a2aa1c0109f583c8d999a995c8b8ce0300e6e..143fd2533065f1ef9e2092fc3d0e85ac7d3ae24d 100644
--- a/third_party/blink/renderer/core/typed_arrays/dom_array_buffer_view.h
+++ b/third_party/blink/renderer/core/typed_arrays/dom_array_buffer_view.h
@@ -39,13 +39,13 @@ class CORE_EXPORT DOMArrayBufferView : public ScriptWrappable {
   ~DOMArrayBufferView() override = default;
 
   DOMArrayBuffer* buffer() const {
-    DCHECK(!IsShared());
+    CHECK(!IsShared());
     DCHECK(dom_array_buffer_);
     return static_cast<DOMArrayBuffer*>(dom_array_buffer_.Get());
   }
 
   DOMSharedArrayBuffer* BufferShared() const {
-    DCHECK(IsShared());
+    CHECK(IsShared());
     DCHECK(dom_array_buffer_);
     return static_cast<DOMSharedArrayBuffer*>(dom_array_buffer_.Get());
   }
